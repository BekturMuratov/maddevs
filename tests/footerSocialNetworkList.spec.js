import {
  mount
} from '@vue/test-utils';
import footerSocialNetworkList from '@/components/Footer/footer-social-network-list';

describe('footer social network list', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(footerSocialNetworkList);
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //
});
