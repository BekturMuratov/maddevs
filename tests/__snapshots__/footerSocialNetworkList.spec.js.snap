// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer social network list renders correctly 1`] = `
<ul
  class="footerSocialNetworkList"
>
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://github.com/maddevsio"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.0274 28.3248C17.0274 28.424 16.9161 28.5034 16.7758 28.5034C16.6161 28.5183 16.5048 28.4389 16.5048 28.3248C16.5048 28.2255 16.6161 28.1461 16.7565 28.1461C16.9016 28.1312 17.0274 28.2106 17.0274 28.3248ZM15.5226 28.1014C15.4887 28.2007 15.5855 28.3148 15.7306 28.3446C15.8565 28.3942 16.0016 28.3446 16.0306 28.2454C16.0597 28.1461 15.9677 28.032 15.8226 27.9873C15.6968 27.9526 15.5565 28.0022 15.5226 28.1014ZM17.6613 28.0171C17.521 28.0518 17.4242 28.1461 17.4387 28.2603C17.4532 28.3595 17.579 28.424 17.7242 28.3893C17.8645 28.3545 17.9613 28.2603 17.9468 28.161C17.9323 28.0667 17.8016 28.0022 17.6613 28.0171ZM20.8452 9C14.1339 9 9 14.2257 9 21.109C9 26.6126 12.3774 31.3222 17.2016 32.9798C17.821 33.0939 18.0387 32.7019 18.0387 32.3793C18.0387 32.0716 18.0242 30.3744 18.0242 29.3322C18.0242 29.3322 14.6371 30.0766 13.9258 27.8533C13.9258 27.8533 13.3742 26.4092 12.5806 26.037C12.5806 26.037 11.4726 25.2578 12.6581 25.2727C12.6581 25.2727 13.8629 25.372 14.5258 26.5531C15.5855 28.4687 17.3613 27.9178 18.0532 27.5903C18.1645 26.7963 18.479 26.2454 18.8274 25.9179C16.1226 25.6102 13.3935 25.2082 13.3935 20.4341C13.3935 19.0693 13.7613 18.3845 14.5355 17.511C14.4097 17.1885 13.9984 15.8585 14.6613 14.1414C15.6726 13.8188 18 15.4813 18 15.4813C18.9677 15.2034 20.0081 15.0595 21.0387 15.0595C22.0694 15.0595 23.1097 15.2034 24.0774 15.4813C24.0774 15.4813 26.4048 13.8138 27.4161 14.1414C28.079 15.8634 27.6677 17.1885 27.5419 17.511C28.3161 18.3894 28.7903 19.0743 28.7903 20.4341C28.7903 25.2231 25.9403 25.6052 23.2355 25.9179C23.6806 26.3099 24.0581 27.0543 24.0581 28.2205C24.0581 29.893 24.0435 31.9624 24.0435 32.3694C24.0435 32.6919 24.2661 33.084 24.8806 32.9699C29.7194 31.3222 33 26.6126 33 21.109C33 14.2257 27.5565 9 20.8452 9ZM13.7032 26.1164C13.6403 26.166 13.6548 26.2801 13.7371 26.3744C13.8145 26.4538 13.9258 26.4886 13.9887 26.424C14.0516 26.3744 14.0371 26.2603 13.9548 26.166C13.8774 26.0866 13.7661 26.0518 13.7032 26.1164ZM13.1806 25.7144C13.1468 25.7789 13.1952 25.8583 13.2919 25.9079C13.3694 25.9576 13.4661 25.9427 13.5 25.8732C13.5339 25.8087 13.4855 25.7293 13.3887 25.6796C13.2919 25.6499 13.2145 25.6648 13.1806 25.7144ZM14.7484 27.4811C14.671 27.5456 14.7 27.6945 14.8113 27.7888C14.9226 27.9029 15.0629 27.9178 15.1258 27.8384C15.1887 27.7739 15.1597 27.625 15.0629 27.5307C14.9565 27.4166 14.8113 27.4017 14.7484 27.4811ZM14.1968 26.7516C14.1194 26.8012 14.1194 26.9302 14.1968 27.0444C14.2742 27.1585 14.4048 27.2082 14.4677 27.1585C14.5452 27.094 14.5452 26.965 14.4677 26.8508C14.4 26.7367 14.2742 26.6871 14.1968 26.7516Z"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://www.facebook.com/maddevsllc"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M33 21.0733C33 14.4054 27.6274 9 21 9C14.3726 9 9 14.4054 9 21.0733C9 27.0995 13.3882 32.0943 19.125 33V24.5633H16.0781V21.0733H19.125V18.4134C19.125 15.3875 20.9165 13.7162 23.6576 13.7162C24.9705 13.7162 26.3438 13.952 26.3438 13.952V16.9231H24.8306C23.3399 16.9231 22.875 17.8538 22.875 18.8086V21.0733H26.2031L25.6711 24.5633H22.875V33C28.6118 32.0943 33 27.0995 33 21.0733Z"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://twitter.com/MadDevsIO"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.5477 30.505C25.6048 30.505 30.5577 23.0016 30.5577 16.495C30.5577 16.2819 30.5534 16.0697 30.5438 15.8585C31.5075 15.1612 32.3392 14.2978 33 13.3087C32.1177 13.7009 31.1683 13.9649 30.1724 14.084C31.1889 13.4743 31.9694 12.51 32.3376 11.3605C31.371 11.9339 30.3135 12.3381 29.2109 12.5557C28.3123 11.5986 27.0332 11 25.6166 11C22.8974 11 20.6924 13.2052 20.6924 15.9233C20.6924 16.3097 20.7356 16.6856 20.8201 17.0461C16.7277 16.8401 13.0988 14.8808 10.6702 11.9014C10.2327 12.6528 10.0026 13.5069 10.0034 14.3763C10.0034 16.0846 10.8727 17.5927 12.1946 18.475C11.4127 18.4512 10.6479 18.24 9.96458 17.8592C9.96385 17.8799 9.96385 17.9 9.96385 17.9221C9.96385 20.3067 11.6611 22.2977 13.9142 22.7489C13.4911 22.8642 13.0544 22.9224 12.6158 22.9222C12.3052 22.922 11.9953 22.8922 11.6904 22.8334C12.3173 24.7898 14.135 26.2135 16.2902 26.2533C14.6049 27.5743 12.4819 28.361 10.1745 28.361C9.77757 28.361 9.38544 28.3385 9 28.2929C11.1792 29.6897 13.7668 30.5047 16.548 30.5047"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://www.linkedin.com/company/mad-devs"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M29.2831 29.4489H25.7271V23.88C25.7271 22.552 25.7034 20.8425 23.8776 20.8425C22.0254 20.8425 21.7421 22.2894 21.7421 23.7833V29.4485H18.186V17.9966H21.5998V19.5616H21.6476C21.9893 18.9775 22.4829 18.4969 23.0761 18.1712C23.6692 17.8454 24.3396 17.6866 25.0159 17.7117C28.6201 17.7117 29.2846 20.0824 29.2846 23.1666L29.2831 29.4489ZM14.1736 16.4312C13.7655 16.4313 13.3665 16.3103 13.0271 16.0836C12.6877 15.8569 12.4231 15.5347 12.2668 15.1576C12.1106 14.7806 12.0697 14.3657 12.1492 13.9653C12.2288 13.565 12.4252 13.1973 12.7138 12.9086C13.0023 12.62 13.37 12.4234 13.7703 12.3437C14.1706 12.264 14.5855 12.3048 14.9626 12.4609C15.3397 12.617 15.6621 12.8815 15.8889 13.2208C16.1157 13.5601 16.2368 13.959 16.2369 14.3672C16.2369 14.6382 16.1836 14.9065 16.0799 15.1569C15.9763 15.4073 15.8243 15.6348 15.6327 15.8265C15.4411 16.0182 15.2137 16.1702 14.9633 16.274C14.7129 16.3777 14.4446 16.4311 14.1736 16.4312ZM15.9516 29.4489H12.3919V17.9966H15.9516V29.4489ZM31.056 9.00163H10.605C10.1408 8.9964 9.69352 9.17566 9.36144 9.50002C9.02937 9.82439 8.83965 10.2673 8.83398 10.7315V31.2673C8.83946 31.7317 9.02905 32.175 9.36112 32.4997C9.69318 32.8244 10.1406 33.004 10.605 32.999H31.056C31.5213 33.0049 31.9699 32.8258 32.3033 32.5011C32.6367 32.1764 32.8275 31.7326 32.834 31.2673V10.73C32.8273 10.2649 32.6364 9.82149 32.303 9.49713C31.9696 9.17277 31.5211 8.99402 31.056 9.00015"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://www.instagram.com/maddevsio"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 9C17.741 9 17.3323 9.01383 16.0524 9.0722C14.7751 9.13047 13.9028 9.33336 13.1395 9.62995C12.3504 9.93662 11.6812 10.347 11.0141 11.0141C10.347 11.6812 9.93662 12.3504 9.62998 13.1396C9.33332 13.9028 9.13051 14.7751 9.07217 16.0524C9.01379 17.3323 9 17.741 9 21C9 24.259 9.01379 24.6677 9.07217 25.9476C9.13051 27.2249 9.33332 28.0972 9.62998 28.8604C9.93665 29.6495 10.347 30.3188 11.0141 30.9859C11.6812 31.653 12.3504 32.0634 13.1395 32.3701C13.9028 32.6666 14.7751 32.8695 16.0524 32.9278C17.3323 32.9862 17.741 33 21 33C24.259 33 24.6677 32.9862 25.9476 32.9278C27.2249 32.8695 28.0972 32.6666 28.8604 32.3701C29.6495 32.0634 30.3188 31.653 30.9859 30.9859C31.653 30.3188 32.0634 29.6495 32.3701 28.8604C32.6666 28.0972 32.8695 27.2249 32.9278 25.9476C32.9862 24.6677 33 24.259 33 21C33 17.741 32.9862 17.3323 32.9278 16.0524C32.8695 14.7751 32.6666 13.9028 32.3701 13.1396C32.0634 12.3504 31.653 11.6812 30.9859 11.0141C30.3188 10.347 29.6495 9.93662 28.8604 9.62995C28.0972 9.33336 27.2249 9.13047 25.9476 9.0722C24.6677 9.01383 24.259 9 21 9ZM21 11.1621C24.2041 11.1621 24.5837 11.1744 25.849 11.2321C27.019 11.2855 27.6544 11.481 28.0772 11.6453C28.6374 11.863 29.0371 12.123 29.457 12.5429C29.877 12.9629 30.137 13.3626 30.3547 13.9227C30.519 14.3456 30.7145 14.981 30.7679 16.151C30.8256 17.4163 30.8379 17.7959 30.8379 21C30.8379 24.2041 30.8256 24.5837 30.7679 25.849C30.7145 27.019 30.519 27.6544 30.3547 28.0772C30.137 28.6374 29.877 29.0371 29.457 29.457C29.0371 29.877 28.6374 30.137 28.0773 30.3547C27.6544 30.519 27.019 30.7145 25.849 30.7679C24.5839 30.8256 24.2044 30.8379 21 30.8379C17.7956 30.8379 17.4162 30.8256 16.151 30.7678C14.981 30.7145 14.3456 30.519 13.9228 30.3547C13.3626 30.137 12.9629 29.877 12.543 29.457C12.1231 29.0371 11.863 28.6374 11.6453 28.0773C11.481 27.6544 11.2855 27.019 11.2321 25.849C11.1744 24.5837 11.1621 24.2041 11.1621 21C11.1621 17.7959 11.1744 17.4163 11.2321 16.151C11.2855 14.981 11.481 14.3456 11.6453 13.9228C11.863 13.3626 12.1231 12.9629 12.5429 12.543C12.9629 12.123 13.3626 11.863 13.9227 11.6453C14.3456 11.481 14.981 11.2855 16.151 11.2321C17.4163 11.1744 17.7959 11.1621 21 11.1621Z"
          fill="#A4A8B4"
        />
         
        <path
          d="M21 25C18.7908 25 17 23.2092 17 21C17 18.7908 18.7908 17 21 17C23.2092 17 25 18.7908 25 21C25 23.2092 23.2092 25 21 25ZM21 14.8378C17.5967 14.8378 14.8378 17.5967 14.8378 21C14.8378 24.4033 17.5967 27.1622 21 27.1622C24.4033 27.1622 27.1622 24.4033 27.1622 21C27.1622 17.5967 24.4033 14.8378 21 14.8378ZM28.8456 14.5944C28.8456 15.3897 28.2009 16.0344 27.4056 16.0344C26.6104 16.0344 25.9656 15.3897 25.9656 14.5944C25.9656 13.7991 26.6104 13.1543 27.4056 13.1543C28.2009 13.1543 28.8456 13.7991 28.8456 14.5944Z"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://medium.com/maddevs-io"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.4859 21.0002C22.4859 25.9706 18.5715 30 13.7431 30C8.91469 30 5 25.9697 5 21.0002C5 16.0306 8.91439 12 13.7431 12C18.5718 12 22.4859 16.0297 22.4859 21.0002ZM32.077 21.0002C32.077 25.6788 30.1198 29.4731 27.7055 29.4731C25.2911 29.4731 23.3339 25.6788 23.3339 21.0002C23.3339 16.3215 25.2908 12.5272 27.7052 12.5272C30.1195 12.5272 32.077 16.3203 32.077 21.0002ZM36 21.0002C36 25.1911 35.3118 28.5906 34.4625 28.5906C33.6133 28.5906 32.9253 25.192 32.9253 21.0002C32.9253 16.8083 33.6136 13.4097 34.4625 13.4097C35.3115 13.4097 36 16.808 36 21.0002Z"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://www.behance.net/maddevs"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25.5648 13.2928H33.0454V15.1037H25.5648V13.2928ZM26.296 24.8112C26.2713 25.7764 26.6037 26.7158 27.2275 27.4435C27.8951 28.0239 28.7615 28.3133 29.6389 28.2487C30.3225 28.2567 30.998 28.0708 31.583 27.7119C32.0233 27.4386 32.3762 27.0425 32.5972 26.572H35.9226C35.4785 27.9349 34.6082 29.1126 33.4444 29.9253C32.2803 30.6505 30.937 31.0229 29.5722 30.9989C28.5887 31.0032 27.613 30.8212 26.695 30.4621C25.8908 30.1302 25.1653 29.6282 24.5664 28.9892C23.9676 28.3503 23.5092 27.5892 23.2217 26.7563C22.8888 25.8278 22.7308 24.8437 22.756 23.8557C22.7361 22.8902 22.894 21.9294 23.2217 21.023C23.5396 20.1758 24.0149 19.3964 24.6174 18.7254C25.2328 18.0977 25.9614 17.5966 26.7633 17.2493C27.6575 16.8901 28.6108 16.7079 29.5722 16.7125C30.6144 16.6784 31.6486 16.909 32.5813 17.3835C33.4169 17.8004 34.1326 18.4297 34.6589 19.2105C35.2327 20.0044 35.6603 20.8969 35.9226 21.8443C36.0258 22.8305 36.0258 23.8251 35.9226 24.8112H26.2976H26.296ZM27.9094 19.9494C27.5597 20.1402 27.245 20.3892 26.9779 20.6867C26.7347 20.9454 26.5726 21.2736 26.5138 21.6244C26.4418 21.9104 26.3971 22.2028 26.3802 22.4975H32.3985C32.3393 21.6896 32.0106 20.9264 31.467 20.3342C30.8892 19.8226 30.1363 19.5635 29.3719 19.6131C28.8651 19.6057 28.3638 19.721 27.9094 19.9494ZM20.9613 27.6456C20.6211 28.3198 20.1188 28.897 19.4989 29.3222C18.8407 29.7558 18.1151 30.0729 17.3529 30.26C16.5661 30.4702 15.757 30.5834 14.9431 30.5963H6V12.0026H14.7428C15.5551 11.9832 16.3658 12.0721 17.1542 12.2694C17.8215 12.3894 18.4595 12.6398 19.0331 13.0067C19.564 13.343 19.9821 13.8345 20.2301 14.4166C20.4717 15.0859 20.5798 15.799 20.5448 16.512C20.5925 17.3527 20.3588 18.1854 19.8804 18.8758C19.394 19.5419 18.747 20.0723 18.0015 20.4183C19.0188 20.6608 19.9169 21.2542 20.5448 22.0949C21.1187 22.9761 21.4191 24.0093 21.4096 25.0635C21.474 25.9487 21.32 26.8363 20.9613 27.6456ZM14.3947 22.3132H9.98989V27.4936H14.312C14.6919 27.4871 15.0703 27.4419 15.4406 27.361C15.7717 27.2981 16.0856 27.1636 16.3612 26.9666C16.6368 26.7697 16.8677 26.5149 17.0382 26.2195C17.2035 25.8396 17.2925 25.4273 17.3036 25.0134C17.3344 24.6253 17.2785 24.2351 17.1404 23.872C17.0022 23.5089 16.7852 23.1822 16.5057 22.9163C15.8823 22.4934 15.1433 22.2823 14.3947 22.3132ZM15.9747 19.01C16.4516 18.5929 16.7012 17.9688 16.6392 17.3333C16.6496 16.965 16.5813 16.5987 16.4389 16.2598C16.295 15.9914 16.09 15.7619 15.8412 15.5904C15.5697 15.438 15.2787 15.3248 14.9765 15.2541C14.6223 15.2188 14.2656 15.2188 13.9114 15.2541H9.98989V19.6309H14.1292C14.7953 19.6325 15.4422 19.4142 15.9731 19.01H15.9747Z"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
   
  <li
    class="footerSocialNetworkList__social-network-link-wrapper"
  >
    <a
      class="footerSocialNetworkList__social-network-link"
      href="https://dev.to/maddevs"
      target="_blank"
    >
      <svg
        fill="none"
        viewBox="0 0 42 42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.435 18.4441C15.2271 18.2887 15.0188 18.2111 14.8109 18.2111H13.8766V23.8077H14.8114C15.0193 23.8077 15.2277 23.73 15.4355 23.5746C15.6434 23.4193 15.7473 23.1862 15.7473 22.875V19.1438C15.7468 18.833 15.6423 18.5995 15.435 18.4441ZM30.6482 9H11.3518C10.0554 9 9.00321 10.0495 9 11.3464V30.6536C9.00321 31.9505 10.0554 33 11.3518 33H30.6482C31.9452 33 32.9968 31.9505 33 30.6536V11.3464C32.9968 10.0495 31.9446 9 30.6482 9ZM17.2607 22.8852C17.2607 23.8929 16.6387 25.4196 14.67 25.4164H12.1843V16.5525H14.7225C16.6211 16.5525 17.2596 18.0771 17.2602 19.0854L17.2607 22.8852ZM22.6543 18.1355H19.8V20.1938H21.5448V21.7779H19.8V23.8355H22.6548V25.4196H19.3238C18.7259 25.4352 18.2288 24.9627 18.2137 24.3648V17.6625C18.1993 17.0652 18.6723 16.5691 19.2696 16.5541H22.6548L22.6543 18.1355ZM28.2064 24.3118C27.4993 25.9591 26.2323 25.6313 25.665 24.3118L23.6009 16.5546H25.3457L26.9373 22.6468L28.5214 16.5546H30.2668L28.2064 24.3118Z"
          fill="#A4A8B4"
        />
      </svg>
    </a>
  </li>
</ul>
`;
